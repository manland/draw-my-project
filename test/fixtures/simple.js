angular.module('publicApp', ['ngRoute'])
.config(['$routeProvider', function($routeProvider) {
  $routeProvider
    .when('/', {
      templateUrl: 'src/login/login.tpl.html',
      controller: 'LoginController'
    })
    .when('/logged', {
      templateUrl: 'src/logged/logged.tpl.html',
      controller: 'LoggedController'
    })
    .when('/logged/:view', {
      templateUrl: 'src/logged/logged.tpl.html',
      controller: 'LoggedController'
    })
    .otherwise({
      redirectTo: '/'
    });
}]);

angular.module('publicApp').controller('LoginController', 
  function ($scope, $http, $location) {
    $scope.showError = false;
    $scope.onLogin = function() {
      console.log('login');
      var url = '/rest/v1/user/login?login=' + $scope.login + '&password=' + $scope.password;
      $http.get(url).success(function(data) {
        $location.url('logged');
      }).error(function(error) {
        console.log(error);
        $scope.showError = true;
        $scope.error = error;
      });
    };
  }
);

angular.module('publicApp').controller('LoggedController', 

  function($scope, $http, $routeParams) {

    var buildViewPath = function(view) {
      return 'src/logged/' + view + '/' + view + '.tpl.html';
    };

    if($routeParams.view) {
      $scope.currentView = buildViewPath($routeParams.view);
    }
  }
);

angular.module('publicApp').factory('LoggedController5', function($scope, $http, $routeParams) {

    var buildViewPath = function(view) {
      return 'src/logged/' + view + '/' + view + '.tpl.html';
    };

    if($routeParams.view) {
      $scope.currentView = buildViewPath($routeParams.view);
    }
  }
);