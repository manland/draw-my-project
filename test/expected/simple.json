{"nodes":[{"name":"angular/$scope","filepath":"","size":1,"imports":[],"type":"angular"},{"name":"angular/$http","filepath":"","size":1,"imports":[],"type":"angular"},{"name":"angular/$location","filepath":"","size":1,"imports":[],"type":"angular"},{"name":"controller/LoginController","filepath":"test/fixtures/simple.js","size":1651,"imports":["angular/$scope","angular/$http","angular/$location"],"type":"controller"},{"name":"angular/$routeParams","filepath":"","size":1,"imports":[],"type":"angular"},{"name":"controller/LoggedController","filepath":"test/fixtures/simple.js","size":1651,"imports":["angular/$scope","angular/$http","angular/$routeParams"],"type":"controller"},{"name":"factory/LoggedController5","filepath":"test/fixtures/simple.js","size":1651,"imports":["angular/$scope","angular/$http","angular/$routeParams"],"type":"factory"}],"advices":[{"node":{"name":"controller/LoginController","filepath":"test/fixtures/simple.js","size":1651,"imports":["angular/$scope","angular/$http","angular/$location"],"type":"controller"},"name":"LoginController have not same name as filename simple !","gravityLevel":2,"gravity":"hot"},{"node":{"name":"controller/LoggedController","filepath":"test/fixtures/simple.js","size":1651,"imports":["angular/$scope","angular/$http","angular/$routeParams"],"type":"controller"},"name":"LoggedController have not same name as filename simple !","gravityLevel":2,"gravity":"hot"},{"node":{"name":"factory/LoggedController5","filepath":"test/fixtures/simple.js","size":1651,"imports":["angular/$scope","angular/$http","angular/$routeParams"],"type":"factory"},"name":"LoggedController5 have not same name as filename simple !","gravityLevel":2,"gravity":"hot"},{"name":"Your controllers are too important compared to services !","gravityLevel":2,"gravity":"hot"},{"node":{"name":"controller/LoginController","filepath":"test/fixtures/simple.js","size":1651,"imports":["angular/$scope","angular/$http","angular/$location"],"type":"controller"},"name":"controller/LoginController does not end with Ctrl.","gravityLevel":2,"gravity":"hot"},{"node":{"name":"controller/LoggedController","filepath":"test/fixtures/simple.js","size":1651,"imports":["angular/$scope","angular/$http","angular/$routeParams"],"type":"controller"},"name":"controller/LoggedController does not end with Ctrl.","gravityLevel":2,"gravity":"hot"},{"node":{"name":"factory/LoggedController5","filepath":"test/fixtures/simple.js","size":1651,"imports":["angular/$scope","angular/$http","angular/$routeParams"],"type":"factory"},"name":"factory/LoggedController5 does not end with Srv.","gravityLevel":2,"gravity":"hot"},{"node":{"name":"controller/LoginController","filepath":"test/fixtures/simple.js","size":1651,"imports":["angular/$scope","angular/$http","angular/$location"],"type":"controller"},"name":"A controller must import $scope and its service !","gravityLevel":1,"gravity":"significant"},{"node":{"name":"controller/LoggedController","filepath":"test/fixtures/simple.js","size":1651,"imports":["angular/$scope","angular/$http","angular/$routeParams"],"type":"controller"},"name":"A controller must import $scope and its service !","gravityLevel":1,"gravity":"significant"}]}